src/plugins/data/common/search/search_source/search_source.ts

```diff
@@ -838,6 +838,12 @@ export class SearchSource {
       body.fields = filteredDocvalueFields;
     }
 
+    // If we aren't requesting any documents, there isn't any reason to request any field information
+    if (body.size === 0) {
```

Comment:
this will only affect requests, and should make no difference to the result es returns right ?

Reply:
That's correct